ros2doctor/ros2doctor/api/__init__.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import platform
+
+import rosdistro
+
+
+def print_platform_info():
+    """Print out platform information."""
+    # platform info
+    print('System Information')
+    print('system   : ', platform.system())
+    print('Linux distribution   : ', platform.dist())
+    print('Mac OS version       : ', platform.mac_ver())
+    print('release  : ', platform.release())
+    print('processor: ', platform.processor())
+    print('\n')
+
+    # python info
+    print('Python')
+    print('version      : ', platform.python_version())
+    print('compiler     : ', platform.python_compiler())
+    print('build        : ', platform.python_build())
+    print('\n')
+
+
+def print_ros2_reqs():
+    """Print out ROS2 distribution info using `rosdistro`."""
+    distro_name = os.environ.get('ROS_DISTRO').lower()
+    u = rosdistro.get_index_url()
+    i = rosdistro.get_index(u)
```

Comment:
Does this raise any exceptions if it can't access the rosdistro index? If so, maybe this should have a fallback where it says it was unable to get the distribution information.